addons/edi/models/edi_gateway.py

```diff
@@ -216,6 +221,28 @@ def _compute_ssh_host_fingerprint(self):
             else:
                 gw.ssh_host_fingerprint = None
 
+
+    @api.depends("outgoing_server")
+    def compute_outgoing_server_id(self):
+        """Get the outgoing server from the smtp host name"""
+        MailServer = self.env["ir.mail_server"]
+        if self.outgoing_server:
```

Comment:
We should add `else` and clear `outgoing_server_id` value if `self.outgoing_server` is not set.